app/models/spree/order_decorator.rb

```diff
@@ -227,6 +227,22 @@ def total_tax
     (adjustments + price_adjustments).sum &:included_tax
   end
 
+  def tax_adjustments
+    adjustments.with_tax +
+      line_items.includes(:tax_adjustments).map {|li| li.tax_adjustments}.flatten
+  end
+
+  def tax_adjustment_totals
+    Hash[tax_adjustments.group_by(&:label).map do |label, adjustments|
+      [adjustments.first.originator, adjustments.sum(&:amount)]
```

Comment:
Hum... 
- Regarding point 4 in testing, does it mean we should add in the "shipping tax setting" the possibility to choose a tax category instead of putting an amount in the field? Is it something easy? I don't see another way to do actually if we want this to be displayed in this new report...
OR if there is any difficulty with that I'm wondering for our urgent case (the shop opens in a week) as we'll use this report for physical stores that don't have shipping then, can we leave like that and just add a mention "WARNING shipping fees VA not included"? Idea @oeoeaio ?

- Regarding the issue 6. Again, as a short term solution we created a new report and this report can be specifically mentioned as "physical shop tax reports without shipping fee and payment method fee". Because the store who needs it doesn't use any shipping fee or payment method fee, so there will be no adjustment issue. BUT there is still the issue that if I change a product and recalculate fees, the enterprise fees need to be recalculated and the corresponding VAT adjusted... hum... is there a solution / quick workaround here @oeoeaio or @ltrls 
I know we need to find out a solution as you said Rob but just looking for quick workaround ... @ltrls I captured for my own understand the three types of adjustements that can be made in an order below
![screenshot from 2017-02-24 08-26-58](https://cloud.githubusercontent.com/assets/8135376/23294386/6db47a7e-fa6b-11e6-9c5d-fbb30253a77c.png)

- For point 7, do you understand what to do @ltrls ? It's really not clear to me but  maybe it will speak to you...

So just to be clear my issue is that **the hub manage who is paying for having this adjusted open his shop early march**... I see there are here complex issues hard to solve quickly due to spree upgrade issues, but it would be great if we could have a short-term solution... so my question is: **is there short term solution or do I need to tell our userwho is paying for that we are not able to do it and he needs to buy another commercial shop management software?** Which is actually complicated because he wants to use the same system to manage his three pick-up points and his physical store so that he has stock consistency... Rrrrrhhhh...